client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/pattern-page-preview.tsx

```diff
@@ -0,0 +1,34 @@
+import { PatternRenderer } from '@automattic/block-renderer';
+import { encodePatternId } from './utils';
+import type { Pattern } from './types';
+import './pattern-page-preview.scss';
+
+interface Props {
+	title: string;
+	patterns: Pattern[];
+	shouldShufflePosts: boolean;
+}
+
+const PATTERN_PAGE_PREVIEW_ITEM_VIEWPORT_WIDTH = 1080;
+
+const PatternPagePreview = ( { title, patterns, shouldShufflePosts }: Props ) => {
+	return (
+		<div className="pattern-assembler__preview">
+			<div className="pattern-assembler__preview-frame">
+				<div className="pattern-assembler__preview-frame-content">
+					{ patterns.map( ( pattern ) => (
+						<PatternRenderer
+							key={ pattern.ID }
+							patternId={ encodePatternId( pattern.ID ) }
+							viewportWidth={ PATTERN_PAGE_PREVIEW_ITEM_VIEWPORT_WIDTH }
```

Comment:
We might need to pass the viewport height to render the pattern using `vh` correctly

Reply:
I've updated the code to pass the container's height as `viewportHeight`. LMK if that's the intended usage.

Reply:
It might be better to keep it the same as the value we used in the pattern list, i.e. `DEFAULT_VIEWPORT_HEIGHT = 500`. The 350 doesn't look good. But we can also iterate it later 🙂

![image](https://github.com/Automattic/wp-calypso/assets/13596067/e1dbd464-028b-4af7-b128-a3a95547a550)